.github/actions/generate-frontend-env/action.yml

```
@@ -0,0 +1,263 @@
+        if [[ -n "${{ inputs.resolution-method }}" ]]; then
+          export VITE_API_RESOLUTION_METHOD="${{ inputs.resolution-method }}"
+        fi
+
+        # Set build metadata if enabled
+        if [[ "${{ inputs.include-build-metadata }}" == "true" ]]; then
+          export VITE_BUILD_TIMESTAMP="$(date -u '+%Y-%m-%d %H:%M:%S UTC')"
+          export VITE_BUILD_COMMIT="${GITHUB_SHA:-unknown}"
+          export VITE_BUILD_BRANCH="${GITHUB_REF_NAME:-unknown}"
+          export VITE_BUILD_WORKFLOW="${GITHUB_WORKFLOW:-unknown}"
+          export VITE_BUILD_RUN_ID="${GITHUB_RUN_ID:-unknown}"
+          
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)